Convert qenum key tests

hackebrot opened the pull request:
Convert TestCase `QEnumKeyTests` to `pytest`. #10

hackebrot committed:
Extract test_no_metaobj to separate module

hackebrot committed:
Convert test_no_metaobj

hackebrot committed:
Convert test_metaobj

hackebrot committed:
Convert test_add_base

hackebrot committed:
Convert test_int_noklass

hackebrot committed:
Convert test_int

hackebrot committed:
Convert test_unknown

hackebrot committed:
Remove former unittest class and skip test_reconverted

hackebrot committed:
Change containing directory name to debug

nicoddemus commented:
Nice work! :smile:

@The-Compiler feel free to merge if you are OK with this change, but mind that your input in `test_reconverted` is welcome.

The-Compiler commented on qutebrowser/test/utils/debug/test_qenum_key.py:
```diff
@@ -0,0 +1,86 @@
+def test_reconverted():
+    """Test passing a flag value which was re-converted to an enum."""
+
+    pytest.skip(msg="It is not clear what this test is supposed to do")
```
That... is a damn good question :blush:. The only thing I can imagine is this causing an exception in earlier versions, and here I wanted to check there's no exception. I'll test with an older version of `qenum_key` to see what happens :)

The-Compiler commented on qutebrowser/test/utils/debug/test_qenum_key.py:
```diff
@@ -0,0 +1,86 @@
+    with pytest.raises(AttributeError):
+        # Make sure it doesn't have a meta object
+        # pylint: disable=pointless-statement,no-member
+        QStyle.PrimitiveElement.staticMetaObject
```
I just noticed a `assert not hasattr(QStyle.PrimitiveElement, 'staticMetaObject')` would probably be much clearer here (and similiar in `test_metaobj()`).

The-Compiler committed:
Remove test_reconverted from qenum_key tests.

The test didn't really seem to test anything useful, and also uses a QFlags
instead of a QEnum.

The-Compiler committed:
test_qenum_key: Use hasattr() for attribute checks.

The-Compiler committed:
Whitespace adjustments

The-Compiler committed:
Remove now unused imports from test_debug.

The-Compiler commented:
I removed `test_reconverted`, switched the attribute tests to `hasattr` as noted in my comment, did some small style adjustments and merged this! :+1:
